engine/src/test/java/org/camunda/bpm/engine/test/bpmn/event/timer/BoundaryTimerNonInterruptingEventTest.java

```diff
@@ -755,12 +767,125 @@ public void shouldExecuteTimeoutListenerJobOnOrAfterDueDate() {
+    // given
+    MyCycleTimerBean myCycleTimerBean = new MyCycleTimerBean("R2/PT1H");
+    processEngineConfiguration.getBeans().put("myCycleTimerBean", myCycleTimerBean);
+    processEngineConfiguration.setReevaluateTimeCycleWhenDue(true);
+
+    runtimeService.startProcessInstanceByKey("nonInterruptingCycle").getId();
+
+    JobQuery jobQuery = managementService.createJobQuery();
+    assertThat(jobQuery.count()).isEqualTo(1);
+    moveByHours(1); // execute first job
+    assertThat(jobQuery.count()).isEqualTo(1);
+
+    // when bean changed and job is due
+    myCycleTimerBean.setCycle("R2/PT2H");
+    moveByHours(1); // execute second job
+
+    // then a job is due in 2 hours
+    assertThat(jobQuery.count()).isEqualTo(1);
+    assertThat(jobQuery.singleResult().getDuedate())
+        .isEqualToIgnoringMinutes(new Date(ClockUtil.getCurrentTime().getTime() + TWO_HOURS));
+
+    moveByHours(2); // execute first job of the new cycle
+
+    // then a job is due in 2 hours
+    assertThat(jobQuery.singleResult().getDuedate())
+        .isEqualToIgnoringMinutes(new Date(ClockUtil.getCurrentTime().getTime() + TWO_HOURS));
+    assertThat(jobQuery.count()).isEqualTo(1);
+
+    moveByHours(2);  // execute second job of the new cycle => no more jobs
+    assertThat(jobQuery.count()).isEqualTo(0);
+  }
+
+  @Test
+  @Deployment(resources = {TIMER_NON_INTERRUPTING_EVENT})
+  public void shouldReevaluateShorterTimerCycleWhenDue() throws Exception {
+    // given
+    MyCycleTimerBean myCycleTimerBean = new MyCycleTimerBean("R3/PT2H");
+    processEngineConfiguration.getBeans().put("myCycleTimerBean", myCycleTimerBean);
+    processEngineConfiguration.setReevaluateTimeCycleWhenDue(true);
+
+    runtimeService.startProcessInstanceByKey("nonInterruptingCycle").getId();
+
+    JobQuery jobQuery = managementService.createJobQuery();
+    assertThat(jobQuery.count()).isEqualTo(1);
+    moveByHours(2); // execute first job
+    assertThat(jobQuery.count()).isEqualTo(1);
+
+    // when bean changed and job is due
+    myCycleTimerBean.setCycle("R2/PT1H");
+    moveByHours(2); // execute second job
+
+    // then one more job is left due in 1 hours
+    assertThat(jobQuery.count()).isEqualTo(1);
+    assertThat(jobQuery.singleResult().getDuedate())
+        .isEqualToIgnoringMinutes(new Date(ClockUtil.getCurrentTime().getTime() + ONE_HOUR));
+
+    moveByHours(1); // execute first job of the new cycle
+
+     // then a job is due in 1 hours
+    assertThat(jobQuery.singleResult().getDuedate())
+        .isEqualToIgnoringMinutes(new Date(ClockUtil.getCurrentTime().getTime() + ONE_HOUR));
+    assertThat(jobQuery.count()).isEqualTo(1);
+
+    moveByHours(1); // execute second job of the new cycle => no more jobs
+    assertThat(jobQuery.count()).isEqualTo(0);
+  }
+
+  @Test
+  @Deployment(resources = {TIMER_NON_INTERRUPTING_EVENT})
+  public void shouldNotReevaluateTimerCycle() throws Exception {
```

Comment:
🔧 I think this is the correct name for this test.
```suggestion
  public void shouldNotReevaluateTimerCycleIfCycleDoesNotChange() throws Exception {
```